### DISCUSSION TITLE
Tutorial Part 6 Unclear Direction Results in Multiple Bugs

### DISCUSSION THREAD

**Discussion by jeustan** [gatsbyjs/gatsby#37073] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Summary

Part 6 of the tutorial directs us to do a couple things that for me have resulted in runtime errors.

Link to the repository - [here](https://github.com/jeustan/first-gatsby-site)

The first bug occurs when a page is created from a template using {mdx.frontmatter__slug}.js. The result is Error in function useStaticQuery in ./.cache/static-query.js:83. I worked around this by removing site meta query in seo and layout and instead hardcoded the page title. I'm pretty new at this so I don't understand why it's throwing this on the programatically created pages and not the others.

<img width="867" alt="Screen Shot 2022-11-18 at 3 01 45 PM" src="https://user-images.githubusercontent.com/39281591/202792558-9a9c6f29-524c-43cd-ac8f-e28f68baaf84.png">

<img width="1085" alt="Screen Shot 2022-11-18 at 3 02 12 PM" src="https://user-images.githubusercontent.com/39281591/202792623-ed836622-58d8-4f52-9cb7-1d50c76c1af1.png">

This bug does not occur on the gatsby cloud hosted run, however— [here](https://firstgatsbysitemain65907.gatsbyjs.io/)

The other bug occurs when we're directed to console.log the props of BlogPost in the blue section "Want to take a closer look?" No example of how to do this is given and anywhere I try to print the props they come back undefined. I have not found any relevant solution to this.

<img width="785" alt="Screen Shot 2022-11-18 at 2 34 16 PM" src="https://user-images.githubusercontent.com/39281591/202789610-c98815ae-50f0-4fdb-a0ec-3ab8a00b1704.png">

I've tried updated packages, gatsby clean, and reinstalling gatsby. I've looked over my code for mistakes (most of it is copy pasted from the tutorial anyways) and haven't seen anything. Any help appreciated, just trying to learn.

### Steps to Resolve this Issue

1. Follow tutorial to part 6.
2.  Launch code in development.
...

**Comment by LekoArts** [comment]
Hi!

About your first problem:

Have you tried stopping the server, running `gatsby clean`? I can't reproduce it with your provided repository.

About your second problem:

You'd add the `console.log` inside the component but before the render function, e.g. here: https://github.com/gatsbyjs/tutorial-example/blob/main/src/pages/blog/%7Bmdx.frontmatter__slug%7D.js#L9

**Reply by jeustan** [reply]
Yes, my post included a list of things I have tried, clean being one of them.

**Reply by jeustan** [reply]
To the second point, that is the first place I placed the console.log but it comes back Uncaught TypeError: Cannot read properties of undefined (reading 'props'). 

```
`const BlogPost = () => {
  console.log(this.props)
  
  return (
    <Layout pageTitle='Blog Posts'>
      <p>My blog post content will go here eventually.</p>
    </Layout>
  )
}

export const Head = () => <Seo title="Super Cool Blog Posts" />

export default BlogPost`
```

<img width="1405" alt="Screen Shot 2022-11-22 at 4 38 14 PM" src="https://user-images.githubusercontent.com/39281591/203426394-056ecc74-61bf-4e5d-abb0-577ea242924c.png">

**Reply by jeustan** [reply]
I cloned the repository at https://github.com/gatsbyjs/tutorial-example and am having the same issues with useStaticQuery but not the console.log. Maybe there's an issue with dependencies?

**Reply by LekoArts** [reply]
It has to be

```jsx
const BlogPost = (props) => {
  console.log(props)
  // Rest...
}
```

Try a clean install with `npm install --legacy-peer-deps`

**Reply by kidCaulfield** [reply]
`Error in function useStaticQuery in ./.cache/static-query.js:83` has happened to me as well. Right after step 2 under *Task: Create blog post page template* in part 6 of the tutorial.

I have used `gatsby clean`, deleted the node modules and the `package-lock.json` the clean installed. Even with @LekoArts suggestion to use `--legacy-peer-deps` option, but it's still broken.

**Reply by kidCaulfield** [reply]
I noticed that the website still runs, once you close the error window you can continue the tutorial @jeustan.

**Reply by ShaneMcX** [reply]
I also repro this issue after directly following the instruction in the tutorial. gatsby clean and restart don't help. Dismissing the error window just shows a blank browser window for the dynamic page.

**Comment by sedsimon** [comment]
Having the exact same problem re: useStaticQuery. Removing the query and hard coding stopped the error from happening. Queries are working fine on non-programatically created pages. `gatsby clean`  and remove `node_modules` did not help.

**Reply by sedsimon** [reply]
...though I also see the problem in gatsby cloud. Error log below:
```
Page data from page-data.json for the failed page /yet-another-post/: {
  "componentChunkName": "component---src-pages-mdx-frontmatter-slug-js-content-file-path-usr-src-app-www-blog-yet-another-post-mdx",
  "path": /yet-another-post/,
  "result": {
    "pageContext": {
      "id": "328f87b6-d63e-55e7-a5fc-4f8b097e5890",
      "frontmatter__slug": "yet-another-post",
      "__params": {
        "frontmatter__slug": "yet-another-post"
      },
      "frontmatter": {
        "title": "Yet Another Post",
        "date": "2021-07-25",
        "slug": "yet-another-post"
      }
    }
  },
  "staticQueryHashes": [],
  "slicesMap": {}
}
```

**Comment by ShaneMcX** [comment]
Not sure if this is related, but I got a bunch of warnings when installing @mdx-js/react:
<img width="941" alt="image" src="https://user-images.githubusercontent.com/16342403/205467961-879d2e13-6919-4473-b738-7b70e7d0f5b4.png">

**Reply by ShaneMcX** [reply]
Repo here: https://github.com/ShaneMcX/gatsby-tutorial

Note also that cloud build fails:

https://www.gatsbyjs.com/dashboard/bef6c072-ea12-473f-9104-620c7568d5af/sites/b1977d35-eec5-4609-bfae-f7275b016853/builds/496a2a1e-29e3-4b40-8c80-fe6b17e69b86/details#rawLogs

![image](https://user-images.githubusercontent.com/16342403/205469594-0810bcaf-7192-49c9-a7e4-c71d59b54518.png)

**Reply by ShaneMcX** [reply]
It did create the dynamic pages: 
![image](https://user-images.githubusercontent.com/16342403/205468184-4a013b7f-b193-4c8f-8d56-d5265a66ccd9.png)

**Reply by ShaneMcX** [reply]
I logged the props in {mdx.frontmatter__slug} and they do appear to contain the correct pageContext results.
![image](https://user-images.githubusercontent.com/16342403/205468459-3bf4c78b-d6ed-41df-a9c2-a45d459c247d.png)

**Reply by ShaneMcX** [reply]
Removing the Layout reference (changing to `<div>`) and commenting out the seo.js include and reference clears the problem: 

```javascript
const BlogPost = (props) => {
  console.log(props);
  return (
    <div pageTitle="Super Cool Blog Posts">
      <p>{props.pageContext.frontmatter.slug}</p>
    </div>
  )
}
```

**Reply by ShaneMcX** [reply]
Restoring the `<Layout>` reference to the template, but commenting out the static query in layout.js also avoids the problem (if seo is also not used).

```javascript
const Layout = ({ pageTitle, children }) => {
  /*
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
        }
      }
    }
  `)
*/
  return (
    <div className={container}>
      <header className={siteTitle}>Temporary Title</header>
      <nav>
```

**Reply by ShaneMcX** [reply]
So it's the static queries in layout.js and seo.js that trigger the problem (if either is present the runtime error is hit, but if neither is used, it's fine).

**Reply by ShaneMcX** [reply]
I completed the tutorial with the hack in place (queries commented out, static text), then added them back in and everything is fine. 🤷
